studio/lib/helpers.ts

```diff
@@ -171,24 +171,30 @@ export async function passwordStrength(value: string) {
   let strength = 0
 
   if (value && value !== '') {
-    const response = await post(`${API_URL}/profile/password-check`, { password: value })
-    if (!response.error) {
-      const { result } = response
-      const score = (PASSWORD_STRENGTH as any)[result.score]
-      const suggestions = result.feedback?.suggestions ? result.feedback.suggestions.join(' ') : ''
-
-      // set message :string
-      message = `${score} ${suggestions}`
-
-      // set strength :number
-      strength = result.score
-
-      // warning message for anything below 4 strength :string
-      if (result.score < DEFAULT_MINIMUM_PASSWORD_STRENGTH) {
-        warning = `${
-          result?.feedback?.warning ? result?.feedback?.warning + '.' : ''
-        } You need a stronger password.`
+    if (value.length < 99){
```

Comment:
```suggestion
    if (value.length <= 99) {
```

Reply:
This function is also getting a bit unreadable with all the nested if statements. Perhaps it would be cleaner to refactor it to "return early". something like

```ts

type PasswordCheckResponse = {
  message: string
  warning: string | null
  strength: number
}

export async function passwordStrength(value: string): Promise<PasswordCheckResponse> {
  if (!value || value === '' || value.length < 6) {
    return {
      message: 'TODO: MESSAGE',
      warning: 'TODO: WARNING',
      strength: 0,
    }
  }

  if (value.length >= 99) {
    return {
      message: 'TODO: MESSAGE',
      warning: 'TODO: WARNING',
      strength: 0,
    }
  }

  const { result, error } = await post(`${API_URL}/profile/password-check`, { password: value })
  const score = (PASSWORD_STRENGTH as any)[result.score]
  const suggestions = result.feedback?.suggestions ? result.feedback.suggestions.join(' ') : ''

  if (error) {
    return {
      message: 'TODO: MESSAGE',
      warning: 'TODO: WARNING',
      strength: 0,
    }
  }

  // warning message for anything below 4 strength :string
  if (result.score < DEFAULT_MINIMUM_PASSWORD_STRENGTH) {
    let warning = `${
      result?.feedback?.warning ? result?.feedback?.warning + '.' : ''
    } You need a stronger password.`

    return {
      message: `${score} ${suggestions}`,
      warning,
      strength: result.score,
    }
  }

  return {
    message: `${score} ${suggestions}`,
    warning: null,
    strength: result.score,
  }
}

```

Reply:
Updated the if check for the value.length to show the error for >= 100 characters, which I think is the same as the suggestion above.  (Just flipped it around for the early returns.)

Refactored the code to return early.  I kept `message`, `warning`, and `strength` variables all declared early to just have the default values for returning in the error cases, instead of defining the literals multiple times.  I can change this to return literals instead if that's preferred.

```js
let message = ''
let warning = ''
let strength = 0

if (!value) {
  return {
    message,
    warning,
    strength,
  }
}
```


```js
if (!value) {
  return {
    message: '',
    warning: '',
    strength: 0,
  }
}
```